src/main/java/io/vavr/collection/Traversable.java

```diff
@@ -1602,31 +1602,48 @@ default Number sum() {
-     * @return A pair of set containing elements split by unzipper
-     * @throws NullPointerException if {@code unzipper} is null
+     * @param unzipper1 a function which converts elements of this to elements of first set
+     * @param unzipper2 a function which converts elements of this to elements of second set
+     * @param <T1>      element type returned by unzipper1
+     * @param <T2>      element type returned by unzipper2
+     * @return A pair of set containing elements split by unzipper1, unzipper2
+     * @throws NullPointerException if any of {@code unzipper1}, {@code unzipper2} are null
      */
-    <T1, T2> Tuple2<? extends Traversable<T1>, ? extends Traversable<T2>> unzip(
-            Function<? super T, Tuple2<? extends T1, ? extends T2>> unzipper);
+    default <T1, T2> Tuple2<Iterator<T1>, Iterator<T2>> unzip(
+            Function<? super T, ? extends T1> unzipper1, Function<? super T, ? extends T2> unzipper2) {
+        Objects.requireNonNull(unzipper1, "unzipper1 is null");
+        Objects.requireNonNull(unzipper2, "unzipper2 is null");
+        final Iterator<T1> iter1 = iterator().map(unzipper1);
+        final Iterator<T2> iter2 = iterator().map(unzipper2);
+        return Tuple.of(iter1, iter2);
+    }
 
     /**
-     * Unzips this elements by mapping this elements to triples which are subsequently split into three distinct
-     * sets.
-     *
-     * @param unzipper a function which converts elements of this to pairs
-     * @param <T1>     1st element type of a triplet returned by unzipper
-     * @param <T2>     2nd element type of a triplet returned by unzipper
-     * @param <T3>     3rd element type of a triplet returned by unzipper
-     * @return A triplet of set containing elements split by unzipper
-     * @throws NullPointerException if {@code unzipper} is null
-     */
-    <T1, T2, T3> Tuple3<? extends Traversable<T1>, ? extends Traversable<T2>, ? extends Traversable<T3>> unzip3(
-            Function<? super T, Tuple3<? extends T1, ? extends T2, ? extends T3>> unzipper);
+     * Unzips this elements by mapping this elements to three distinct sets.
+     *
+     * @param unzipper1 a function which converts elements of this to first set
+     * @param unzipper2 a function which converts elements of this to second set
+     * @param unzipper3 a function which converts elements of this to third set
+     * @param <T1>      element type returned by unzipper1
+     * @param <T2>      element type returned by unzipper2
+     * @param <T3>      element type returned by unzipper3
+     * @return A triplet of {@code Iterator} to iterate over elements of sets split by {@code unzipper1}, {@code unzipper2}, {@code unzipper3}
+     * @throws NullPointerException if any of {@code unzipper1}, {@code unzipper2}, {@code unzipper3} is null
+     */
+    default <T1, T2, T3> Tuple3<? extends Iterator<T1>, ? extends Iterator<T2>, ? extends Iterator<T3>> unzip3(
```

Comment:
@danieldietrich , should this be 
`... Tuple3<? extends Iterator<T1>, ? extends Iterator<T2>, ? extends Iterator<T3>> unzip3(...) `
or
`... Tuple3<Iterator<T1>, Iterator<T2>, Iterator<T3>> unzip3(...)`?

Same question goes to `unzip(...)` above, line 1614.

Reply:
Please do it like this (for return types in general):

```java
Tuple3<Iterator<T1>, Iterator<T2>, Iterator<T3>>
```

Thank you!